cmd/permissions-checker/main.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+)
+
+type defaultPermissions struct {
+	file  string
+	dir   string
+	owner string
+	group string
+}
+
+type searchSettings struct {
```

Comment:
please document this structure and the fields. e.g. what is "exclude"?
if it's a regexp, let's make this field be a `*regexp.Regexp` (by compiling the regexp before creating a value of this structure)